crates/core/src/node/inner/blockchain.rs

```diff
@@ -620,6 +635,26 @@ impl BlockchainState {
         self.blocks.insert(block.hash, block);
     }
 
+    pub(super) fn apply_batch(&mut self, tx_results: impl IntoIterator<Item = TransactionResult>) {
+        self.current_batch += 1;
+        // Our version of contracts has system log and data availability verification disabled.
+        // Hence, we are free to create a zeroed out dummy header without any logs whatsoever.
+        let header = L1BatchHeader::new(
+            self.current_batch,
+            0,
+            // We could use contract hashes that were actually used, but they might differ from what
+            // L1 expects due to impersonation
```

Comment:
If a user checkouts different system contracts that has system log and data availability verification enabled will that be a problem? Just thinking about our own protocol devs usage.

Reply:
I have slightly tweaked this place to use real timestamp and contract hashes as I think `core` should not be worried with what is happening in `l1-sidecar`. Hopefully less confusion going forward, `l1-sidecar` has an equivalent comment anyway (where it overrides batch's contract hashes and fee address).

To answer your question: we intentionally deploy "constant" contracts on L1 as per `./l1-setup`. They have nothing to do with what user might provide in runtime (i.e. there is no way to override those from a regular user's standpoint). However, it is possible to re-generate anvil state with a different set of contracts by following the guide in `./l1-setup`. Protocol devs should be able to handle that but admittedly the flow is a bit clunky.